cmd/controller-manager/app/controller_manager.go

```diff
@@ -90,19 +90,20 @@ func Run(controllerManagerOptions *options.ControllerManagerServer) error {
 	// Build the K8s kubeconfig / client / clientBuilder
 	glog.V(4).Info("Building k8s kubeconfig")
 
-	k8sKubeconfig, err := rest.InClusterConfig()
+	var err error
+	var k8sKubeconfig *rest.Config
+	if controllerManagerOptions.K8sAPIServerURL == "" && controllerManagerOptions.K8sKubeconfigPath == "" {
+		k8sKubeconfig, err = rest.InClusterConfig()
+	} else {
+		k8sKubeconfig, err = clientcmd.BuildConfigFromFlags(
+			controllerManagerOptions.K8sAPIServerURL,
+			controllerManagerOptions.K8sKubeconfigPath)
```

Comment:
It should be possible.  The factory returns [this](https://github.com/kubernetes/kubernetes/blob/c7d204cbfc2d39e436579b096e435da657f73326/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L61) interface, so something like this might work:

```go
util.DefaultClientConfig(flags).ClientConfig()
```

Also, these lines from the Factory look interesting and could be enough (I haven't looked deeply).  Both those are present in client-go, so if they worked there would be no need to vendor in kubectl code:

```go
loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
...
flags.StringVar(&loadingRules.ExplicitPath, "kubeconfig", "", "Path to the kubeconfig file to use for CLI requests.")
```

Probably out of Out of scope for this PR